mitodl/ocw-to-hugo
# youtube shortcode refactor

#### Pre-Flight checklist


#### What are the relevant tickets?
Closes https://github.com/mitodl/ocw-to-hugo/issues/210
Closes https://github.com/mitodl/ocw-to-hugo/issues/207

#### What's this PR do?
This PR refactors the way `youtube` shortcodes are generated by instead replacing the Youtube embed codes with a div with a special class.  This class of div is then caught in a turndown rule and replaced with the proper shortcode.  The `media_location` property is deposited into the div as the content and is pulled out in the turndown rule with `node.textContent` so it is not run through markdown transformation.

Note that some other general refactoring is being done in this PR.  All test data has been updated with the latest parsed json, some unused imports were removed and there was a failing test after updating the test json that was corrected.

#### How should this be manually tested?
Spin up a local test site using `ocw-course-hugo-theme` (read the readme if you've never done this before) for `8-01sc-classical-mechanics-fall-2016` and visit http://localhost:1313/sections/week-1-kinematics/3.1-coordinate-system-and-position-vector-in-2d/.  The YouTube video should now properly render.

#### Screenshots (if appropriate)
![image](https://user-images.githubusercontent.com/12089658/109566531-01544780-7ab2-11eb-93d4-c952a631b1a8.png)
